### ISSUE TITLE
CRACO 7 / CRA 5 / Types can't be found

### ISSUE THREAD

**Issue by Snotax** [arthurfiorette/axios-cache-interceptor#430]
### Describe the bug

TS7016: Could not find a declaration file for module 'axios-cache-interceptor'. 'C:/.../packages/app/node_modules/axios-cache-interceptor/dist/index.cjs' implicitly has an 'any' type.
  Try `npm i --save-dev @types/axios-cache-interceptor` if it exists or add a new declaration (.d.ts) file containing `declare module 'axios-cache-interceptor';`

### To Reproduce
CRACO 7 / CRA 5 / React 18

CRACO config:
```
module.exports = {
  webpack: {
    configure: (webpackConfig) => {
      // Allow importing from external workspaces.
      monorepoWebpackTools.enableWorkspacesResolution(webpackConfig);
      // Ensure nohoisted libraries are resolved from this workspace.
      monorepoWebpackTools.addNohoistAliases(webpackConfig);

      webpackConfig.devtool = 'source-map';

      webpackConfig.module.rules.push({
        test: /\.mjs$/,
        include: /node_modules/,
        type: "javascript/auto"
      });

      webpackConfig.resolve.plugins.find((plugin) => plugin.constructor.name == "ModuleScopePlugin").appSrcs = workspaces;

      webpackConfig.resolve.mainFields = ['browser', 'module', 'main'];

      return webpackConfig;
    },

    plugins: [
      // Inject the "__DEV__" global variable.
      new webpack.DefinePlugin({
        __DEV__: process.env.NODE_ENV !== "production",
      })
    ],
  },
};

```

### Expected behavior
 Types would be found

### Additional context
- Axios Cache Interceptor:latest
- Node Version: 19.2.0

Do you have an idea why craco can't find your types?

**Comment by arthurfiorette** [comment]
Hey, I do not have sufficient experience with your setup to be helpful here. But maybe it is because of the custom `.cjs` extension we are using?

**Comment by vtereshyn** [comment]
@arthurfiorette I can confirm, that typings are not visible in Typescript when using axios-cache-interceptor in ESM files

**Comment by Snotax** [comment]
Would be nice to get the correct types for ESM in here,  I really liked the package :D

**Comment by arthurfiorette** [comment]
Hello, sorry for the delayed response. 

Fixed it and it will be shipped on the 1.0.0 release.
